src/components/AutoComplete/AutoComplete.js

```diff
@@ -25,42 +25,52 @@ function renderMenu(items, value, style) {
 }
 
 export default function AutoComplete(props) {
-  const { showClear, source, 'data-testid': testid, id } = props;
+  const {
+    showClear,
+    source,
+    'data-testid': testid,
+    id,
+    clickedTag,
+    onSelect,
+  } = props;
   const [value, setValue] = useState('');
 
-  const setPermalinkParams = (param, value) => {
-    const permalink = new URLSearchParams(window.location.search);
-    const paramItem = source.filter(item => item.label === value);
-    if (paramItem.length && value.length) {
-      permalink.set(param, paramItem[0].value);
-    } else if (!value.length) {
-      permalink.delete(param);
-    }
-    window.history.pushState({}, null, '?' + permalink.toString());
-  };
+  const setPermalinkParams = useCallback(
```

Comment:
`useCallback` 👍

Reply:
the file still needs a refactor, but I'm waiting for the filter to hooks to get merged, I think we should move the state to the upper level or implement a global store/state solution

Reply:
#559